### PR TITLE
Make sure 'new' terms don't exist before `wp_insert_term`

### PR DESCRIPTION
**Pull request by mgratch**
This might not be the most performant but it solves the problem. Honestly I don't know if there is a more efficient answer.

### CODE REVIEW

File: classes/PodsAPI.php [hunk 1]
```diff
@@ -701,7 +701,23 @@ public function save_term( $term_data, $term_meta, $strict = false, $sanitized =
 
 			unset( $term_data['name'] );
 
-			$term_data['term_id'] = wp_insert_term( $term_name, $taxonomy, $term_data );
+			// Check to be certain the 'new' term doesn't exist.
+			if ( is_int( $term_name ) ) {
```
Comment (sc0ttkclark) [inline classes/PodsAPI.php]: Is this necessary for your use-case?

File: classes/PodsAPI.php [hunk 2]
```diff
@@ -701,7 +701,23 @@ public function save_term( $term_data, $term_meta, $strict = false, $sanitized =
...
+				$term = get_term_by( 'term_id', $term_name, $taxonomy );
+			} else {
+				$term = get_term_by( 'name', $term_name, $taxonomy );
+
+				if ( ! $term ) {
+					$term = get_term_by( 'slug', $term_name, $taxonomy );
```
Comment (sc0ttkclark) [inline classes/PodsAPI.php]: Is this necessary for your use-case?

File: classes/PodsAPI.php (hunk 1 above)
Comment (sc0ttkclark) [reply classes/PodsAPI.php]: @mgratch

File: classes/PodsAPI.php (hunk 2 above)
Comment (sc0ttkclark) [reply classes/PodsAPI.php]: @mgratch

Comment (mgratch) [reply classes/PodsAPI.php]: So after reviewing this further, this PR seems unnecessary. Core adds terms by the same process and doesn't do any last minute checks.  Validation should happen before this, in my case in another plugin.

**Comment by pglewis** [comment]
What's the best label here, "out of scope" @mgratch?  I'll label it and close it out

**Comment by pglewis** [comment]
Also, I assume the related issue #4770 can be closed as well?

**Comment by mgratch** [comment]
Yes, the issue can be closed as well :)

**Comment by sc0ttkclark** [comment]
Thanks!

**Pull request closed without merging**